model/testing/src/icon4py/model/testing/serialbox.py

```diff
@@ -1874,6 +2017,62 @@ def from_savepoint_velocity_init(
             savepoint, self.serializer, size=self.grid_size, backend=self.backend
         )
 
+    def savepoint_velocity_1_7_compute_edge_diagnostics_for_velocity_advection_init(
```

Comment:
why not call it simply
```suggestion
    def compute_edge_diagnostics_for_velocity_advection_init(
```
or at least invert to have the `savepoint` stuff only at the end. 
Also the name of the Savepoint classes could be changed if you have a better idea or keep them as is to match the savepoint name.